### PR TITLE
q42

### PR DESCRIPTION
**Pull request by Navleenkaursandhu**
q42:
image

### CODE REVIEW

File: Javascript/q42.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      width: 200px;
+    }
+
+    button {
+      margin-top: 5px;
+    }
+  </style>
+
+  <script>
+    function myFunction(){
```
Comment (Mantaseus) [inline Javascript/q42.html:31]: This is not a very descriptive function name. Please improve this

File: Javascript/q42.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      let min = Number(document.getElementById("demo0").value);
+      let max = Number(document.getElementById("demo1").value);
+      let x = Math.round(Math.random() * (max - min + 1) + min);
```
Comment (Mantaseus) [inline Javascript/q42.html:34]: What is `x`? Please improve this variable name

File: Javascript/q42.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      document.querySelector("#demo").innerHTML = " " + x;
```
Comment (Mantaseus) [inline Javascript/q42.html:35]: What is the `" "` for? Did you try putting a space between the `Generated Num:` and the `<span` definition in the HTML below?

File: Javascript/q42.html (hunk 2 above)
Comment (Mantaseus) [inline Javascript/q42.html:34]: Are you sure this function will always be correct? For `max` of 1 and a `min` of 0, this function might some times give us `2` as well. Please fix

File: Javascript/q42.html [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+<body>
+  <div class="flex">
+    <div>Min limit</div>
+    <input id="demo0" type="number" class="box" style="margin-left: 4px;"></input>
+    <div>Max limit</div>
+    <input id="demo1" type="number" class="box"></input>
+  </div>
+  <button onclick="myFunction()">Generate random</button>
+  <div>Generated Num:<span id="demo"></span></div>
```
Comment (Mantaseus) [inline Javascript/q42.html:48]: Please fix all of these `id` values. `demo` is not a very useful or descriptive name.

Comment (Mantaseus) [reply Javascript/q42.html:48]: Please fix them in the `input` elements above as well

File: Javascript/q42.html (hunk 1 above)
Comment (Navleenkaursandhu) [reply Javascript/q42.html:31]: done the changes

File: Javascript/q42.html (hunk 2 above)
Comment (Navleenkaursandhu) [reply Javascript/q42.html:34]: done the changes

File: Javascript/q42.html (hunk 1 above)
Comment (Navleenkaursandhu) [reply Javascript/q42.html:31]: ok

File: Javascript/q42.html (hunk 3 above)
Comment (Navleenkaursandhu) [reply Javascript/q42.html:35]: oh yes, done

File: Javascript/q42.html (hunk 2 above)
Comment (Navleenkaursandhu) [reply Javascript/q42.html:34]: ok, have used Math.floor() to fix this issue, please suggest whether its correct or not

File: Javascript/q42.html (hunk 4 above)
Comment (Navleenkaursandhu) [reply Javascript/q42.html:48]: fixed

File: Javascript/q42.html (hunk 2 above)
Comment (Mantaseus) [reply Javascript/q42.html]: Yeah, that works

**Review by Mantaseus — APPROVED** [review]

**Pull request merged**